File: benchmarks/python/format_str_int.py
def format_str_int_bench(iterations: int) -> int:
    sink = 0
    i = 0
    while i < iterations:
        s = str(i)
        sink += len(s) + ord(s[-1])
        i += 1
    return sink


def main() -> int:
    iterations = 400000
    result = format_str_int_bench(iterations)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: benchmarks/rust/format_str_int.rs
fn format_str_int_bench(iterations: i64) -> i64 {
    let mut sink: i64 = 0;
    let mut i: i64 = 0;
    while i < iterations {
        let s = i.to_string();
        sink += s.len() as i64 + s.as_bytes()[s.len() - 1] as i64;
        i += 1;
    }
    sink
}

fn main() {
    let iterations: i64 = 400_000;
    let result = format_str_int_bench(iterations);
    println!("{}", result);
}
